### DISCUSSION TITLE
Networking Woes

### DISCUSSION THREAD

**Discussion by Chireven** [OSDeploy/OSD#185] [General]
Greetings:

I have an HP ZBook Firefly 16" G9 that I'm trying to deploy Windows 11 24H2 to.  In the PE Stage, I'm having troubles getting a network address from DHCP.  I've tried with a belkin (realtek) F2CU040 and the HP Dock (also realtek), and neither work.  

The IP address is always an APIPA address (169.254.0.0/16).  Attempts to renew the DHCP address produce the following error:
"An error occurred while renewing interface Ethernet : The data is invalid".  

I've found various versions of the driver and manually loaded them with drvload with no changes to this behavior.   I built the media using the following process:

1. Installed ADK
2. Installed WinPE Addon
3. > install-module osd -force
4. > new-osdcloudtemplate 
5. > new-osdcloudworkspace -workspacepath d:\osdcloud\MyWorkspace
6. Tried various manipulations of Cloud Drivers 

- > edit-osdcloudwinpe -clouddriver HP
- > edit-osdcloudwinpe -clouddriver Dell, HP
- > edit-osdcloudwinpe -clouddriver *
- > edit-osdcloudwinpe -clouddriver HP -driverpath d:\osdcloud\drivers ( this is the path I put drivers I was testing in to bring them into PE)

7. > new-osdcloudUSB

Upon booting, the media repeats the error that it 'IP Address not yet assigned by DHCP, trying to get a new DHCP lease..".  The script eventually continues and shows two network adapters -- an Intel Nic (which I cannot use, it's wifi) and the realitek USB-C nic (Realtek USB GbE Family Controller).

I've also repeated everything above with the configuration setting to pass through the device mac enable and disabled.

Everything looks as though the device and driver are working fine, but I cannot for the life of me get this thing to pick up a DHCP lease.  I am confident this isn't a problem with this project, but I'm wondering if anyone here may have had a similar issue or know of any "known issues" using realtek cards in WinPE.

Appreciate the help!

**Comment by Chireven** [comment]
I went back a few versions of WinPE, to Win10-2004, and everything just started working fine.  Not sure I want to stay at this version for deployments, but at least it gets me moving in the right direction!

If anyone stumbles across this and needs the adk: https://learn.microsoft.com/en-us/windows-hardware/get-started/adk-install#other-adk-downloads

**Comment by AkosBakos** [comment]
You can see here & check the basic setup below: https://github.com/OSDeploy/WPNinjas2024/tree/main/1%20Explore%20OSDCloud
